### PR TITLE
[DIGITALRIV-28][DIGITALRIV-27] Q3 App Review

### PR DESCRIPTION
**Pull request by polishq**
As part of our team's 2022 Q3 App Review, this PR:
- Adds the `@checkAdminAccess` GraphQL directive for improved security
- Moves the Digital River admin navigation link to the Transactions section and adds the payment-related `LMProductId` to control admin user access
- Reduces unnecessary logger usage
- Fixes a bug where the app expected the Digital River order state to be `payment_pending` but the actual state is `pending_payment` (all possible order states are now specified in the typings)
- Re-enables automatic retries on HTTP requests; previously these were disabled to prevent duplicate orders being created if the first attempt timed out. Now retries are disabled only on the `createOrder` method.
- Refactors backend code to address SonarCloud warnings and improve optimization (no more await in loop, for example)

This new version has been published as a beta and installed in `sandboxusdev` master. I have tested tax calculation and successfully placed (and settled) a test order.

### CODE REVIEW

File: node/resolvers/directives/checkAdminAccess.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      throw new AuthenticationError('No Token Was Provided')
+    }
+
+    try {
+      await identity.validateToken({ token: adminUserAuthToken })
+    } catch (err) {
+      logger.warn({
+        error: err,
+        token: adminUserAuthToken,
+        message: 'Avalara-AttemptedUnauthorizedAccess',
```
Comment (Arturo777) [inline node/resolvers/directives/checkAdminAccess.ts:24]: Avalara is related to this logger?

**Review by Arturo777 — CHANGES_REQUESTED** [review]

File: node/resolvers/directives/checkAdminAccess.ts (hunk 1 above)
Comment (polishq) [reply node/resolvers/directives/checkAdminAccess.ts]: Good catch! Serves me right for copying code. Fixed

**Review by whc105 — APPROVED** [review]
LGTM

**Review by Arturo777 — APPROVED** [review]
LOGM

**Pull request merged**